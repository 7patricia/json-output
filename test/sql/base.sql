\set ECHO none
SET client_min_messages TO error;
\t on
\x off
SELECT 'created logical replication slot'
  FROM pg_create_logical_replication_slot('json', 'json-output');

BEGIN;

CREATE SCHEMA IF NOT EXISTS json;
SET LOCAL search_path TO json, public;
CREATE EXTENSION IF NOT EXISTS hstore;

CREATE TABLE IF NOT EXISTS test (
  i   integer PRIMARY KEY,
  h   hstore DEFAULT ''
);

INSERT INTO test (i) SELECT i FROM generate_series(1, 8) AS i;

CREATE VIEW changedata AS
SELECT data FROM pg_logical_slot_get_changes('json', NULL, NULL);

END;

BEGIN;
SET LOCAL search_path TO json, public;
UPDATE test SET i = i * 10 WHERE i % 3 = 0;
UPDATE test SET h = hstore('i', i::text)||hstore('2i', (2*i)::text);
UPDATE test SET h = NULL WHERE i % 4 = 0;
DELETE FROM test WHERE i % 2 = 1;
END;

SELECT * FROM json.changedata;

SELECT 'deleted logical replication slot'
  FROM pg_drop_replication_slot('json');

DROP SCHEMA json CASCADE;
