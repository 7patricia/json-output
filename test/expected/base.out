\set ECHO none
 created logical replication slot

 {"schema":"json","table":"test","action":"insert","new":{"i":1,"h":{}}}
 {"schema":"json","table":"test","action":"insert","new":{"i":2,"h":{}}}
 {"schema":"json","table":"test","action":"insert","new":{"i":3,"h":{}}}
 {"schema":"json","table":"test","action":"insert","new":{"i":4,"h":{}}}
 {"schema":"json","table":"test","action":"insert","new":{"i":5,"h":{}}}
 {"schema":"json","table":"test","action":"insert","new":{"i":6,"h":{}}}
 {"schema":"json","table":"test","action":"insert","new":{"i":7,"h":{}}}
 {"schema":"json","table":"test","action":"insert","new":{"i":8,"h":{}}}
 {"schema":"json","table":"test","action":"update","new":{"i":30,"h":{}},"old":{"i":3,"h":null}}
 {"schema":"json","table":"test","action":"update","new":{"i":60,"h":{}},"old":{"i":6,"h":null}}
 {"schema":"json","table":"test","action":"update","new":{"i":1,"h":{"i": "1", "2i": "2"}}}
 {"schema":"json","table":"test","action":"update","new":{"i":2,"h":{"i": "2", "2i": "4"}}}
 {"schema":"json","table":"test","action":"update","new":{"i":4,"h":{"i": "4", "2i": "8"}}}
 {"schema":"json","table":"test","action":"update","new":{"i":5,"h":{"i": "5", "2i": "10"}}}
 {"schema":"json","table":"test","action":"update","new":{"i":7,"h":{"i": "7", "2i": "14"}}}
 {"schema":"json","table":"test","action":"update","new":{"i":8,"h":{"i": "8", "2i": "16"}}}
 {"schema":"json","table":"test","action":"update","new":{"i":30,"h":{"i": "30", "2i": "60"}}}
 {"schema":"json","table":"test","action":"update","new":{"i":60,"h":{"i": "60", "2i": "120"}}}
 {"schema":"json","table":"test","action":"delete","old":{"i":1,"h":null}}
 {"schema":"json","table":"test","action":"delete","old":{"i":5,"h":null}}
 {"schema":"json","table":"test","action":"delete","old":{"i":7,"h":null}}

 deleted logical replication slot

